build_system/seqan3-config.cmake

```diff
@@ -0,0 +1,527 @@
+include (CheckIncludeFileCXX)
+include (CheckCXXSourceCompiles)
+
+# ----------------------------------------------------------------------------
+# Options for CheckCXXSourceCompiles
+# ----------------------------------------------------------------------------
+
+# deactivate messages in check_*
+set (CMAKE_REQUIRED_QUIET       1)
+# use global variables in Check* calls
+set (CMAKE_REQUIRED_INCLUDES    ${CMAKE_INCLUDE_PATH})
+set (CMAKE_REQUIRED_FLAGS       ${CMAKE_CXX_FLAGS})
+
+# ----------------------------------------------------------------------------
+# Pretty printing and error handling
+# ----------------------------------------------------------------------------
+
+macro (seqan3_config_print text)
+    if (NOT ${FIND_NAME}_FIND_QUIETLY)
+        message (STATUS "  ${text}")
+    endif ()
+endmacro ()
+
+macro (seqan3_config_error text)
+    if (${FIND_NAME}_FIND_REQUIRED)
+        message (FATAL_ERROR ${text})
+    else ()
+        if (NOT ${FIND_NAME}_FIND_QUIETLY)
+            message (WARNING ${text})
+        endif ()
+        return ()
+    endif ()
+endmacro ()
+
+# ----------------------------------------------------------------------------
+# Force-deactivate optional dependencies
+# ----------------------------------------------------------------------------
+
+# Cereal is auto-detected by default, i.e. used if found, not used if not found.
+# You can optionally set a hard requirement so a build fails without cereal,
+# or you can force-disable cereal even if present on the system.
+option (SEQAN3_CEREAL    "Require cereal and fail if not present." OFF)
+option (SEQAN3_NO_CEREAL "Don't use cereal, even if present." OFF)
+
+if (SEQAN3_CEREAL AND SEQAN3_NO_CEREAL)
+    # this is always a user error, therefore we always error-out, even if SeqAn is not required
+    message (FATAL_ERROR "You may not specify SEQAN3_CEREAL and SEQAN3_NO_CEREAL at the same time.\n\
+                          You can specify neither (use auto-detection), or specify either to force on/off.")
+    return ()
+endif ()
+
+if (SEQAN3_CEREAL)
+    set (SEQAN3_DEFINITIONS ${SEQAN3_DEFINITIONS} "-DSEQAN3_WITH_CEREAL=1")
+elseif (SEQAN3_NO_CEREAL)
+    set (SEQAN3_DEFINITIONS ${SEQAN3_DEFINITIONS} "-DSEQAN3_WITH_CEREAL=0")
+endif ()
+
+# These two are "opt-in", because detected by CMake
+# If you want to force-require these, just do find_package(zlib REQUIRED) before find_package(seqan3)
+option (SEQAN3_NO_ZLIB  "Don't use ZLIB, even if present." OFF)
+option (SEQAN3_NO_BZIP2 "Don't use BZip2, even if present." OFF)
+
+# ----------------------------------------------------------------------------
+# Require C++17
+# ----------------------------------------------------------------------------
+
+set (CMAKE_REQUIRED_FLAGS_ORIGINAL ${CMAKE_REQUIRED_FLAGS})
+
+set(CXXSTD_TEST_SOURCE
+    "#if !defined(__cplusplus) || (__cplusplus < 201703L)
+    #error NOCXX17
+    #endif
+    int main() {}")
+check_cxx_source_compiles("${CXXSTD_TEST_SOURCE}" CXX17_BUILTIN)
```

Comment:
I think spaces between these statements make it a little bit more readable.
Like before an if-clause begins, insert a new line.
If a multi line statement ends insert a new line.